### ISSUE TITLE
docs(actions): invalid link caused by an internal link in reusable

### ISSUE THREAD

**Issue by muzimuzhi** [github/docs#25086]
### Code of Conduct

- [X] I have read and agree to the GitHub Docs project's [Code of Conduct](https://github.com/github/docs/blob/main/CODE_OF_CONDUCT.md)

### What article on docs.github.com is affected?

https://docs.github.com/en/actions/using-workflows/triggering-a-workflow

### What part(s) of the article would you like to see updated?

In https://docs.github.com/en/actions/using-workflows/triggering-a-workflow#using-filters-to-target-specific-branches-for-pull-request-events, it reads

> If you define both branches/branches-ignore and [paths](https://docs.github.com/en/actions/using-workflows/triggering-a-workflow#onpushpull_requestpull_request_targetpathspaths-ignore), the workflow will only run when both filters are satisfied.

In another section of the same page https://docs.github.com/en/actions/using-workflows/triggering-a-workflow#using-filters-to-target-specific-branches-or-tags-for-push-events, it reads

> If you define both branches/branches-ignore and [paths](https://docs.github.com/en/actions/using-workflows/triggering-a-workflow#onpushpull_requestpull_request_targetpathspaths-ignore), the workflow will only run when both filters are satisfied.

Here the two "paths" 
- are both linked to the same, non-existent target on the same page\
  https://docs.github.com/en/actions/using-workflows/triggering-a-workflow#onpushpull_requestpull_request_targetpathspaths-ignore
- while the expected target is in another page\
  https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#onpushpull_requestpull_request_targetpathspaths-ignore

### Additional information

This is caused by the use of an internal link in reusable, see line 5 below
```md
[`paths`](#onpushpull_requestpull_request_targetpathspaths-ignore)
```
https://github.com/github/docs/blob/1b901942757229900a2c3374bb27e87d78847b63/data/reusables/actions/workflows/section-triggering-a-workflow-branches.md?plain=1#L5-L7

When one reusable is used by different pages, in most cases internal links like `[text](#target)` will lead to invalid url in at least one of those rendered pages. The correct way is to tell the full path of the target page, like another link in line 7 in snippet above, 
```md
[AUTOTITLE](/actions/using-workflows/workflow-syntax-for-github-actions#filter-pattern-cheat-sheet)
```

I'll provide a PR.

**Comment by cmwilson21** [comment]
@muzimuzhi Thank you for opening an issue and linking it to your PR! ✨ 

And thanks for providing all this extra context! 💖

**Comment by peterbe** [comment]
I did a search for the pattern of in-page anchor links with the `data/reusables/`. Wow, there's a lot!
<img width="813" alt="Screenshot 2023-04-20 at 4 08 41 PM-with-shadow" src="https://github.com/github/docs/assets/26739/40df9a1b-6d64-4240-8af3-91334686d29f">

I wonder if we have a lot of these busted in-page anchor links due to this very situation that @muzimuzhi so rightfully spotted. 

We currently don't have a link checker for anchor links. It's something we've discussed and will continue to discuss. This issue here certainly helps. Thanks @muzimuzhi !

/CC @sarahs
